dask_kubernetes/operator/controller/controller.py

```diff
@@ -381,6 +382,75 @@ async def retire_workers(
         return [w["metadata"]["name"] for w in workers.items[:-n_workers]]
 
 
+async def check_scheduler_idle(scheduler_service_name, namespace, logger):
+    # Try getting idle time via HTTP API
+    dashboard_address = await get_scheduler_address(
+        scheduler_service_name,
+        namespace,
+        port_name="http-dashboard",
+        allow_external=False,
+    )
+    async with aiohttp.ClientSession() as session:
+        url = f"{dashboard_address}/api/v1/check_idle"
+        async with session.get(url) as resp:
+            if resp.status <= 300:
+                idle, idle_since = await resp.json().values()
+                if idle:
+                    logger.debug("Scheduler idle since: %s", idle_since)
+                return idle, idle_since
+            logger.debug(
+                "Received %d response from scheduler API with body %s",
+                resp.status,
+                await resp.text(),
+            )
+
+    # Otherwise try gracefully checking via the RPC
+    logger.info(
+        f"Checking {scheduler_service_name} idleness failed via the HTTP API, falling back to the Dask RPC"
+    )
+    # Dask version mismatches between the operator and scheduler may cause this to fail in any number of unexpected ways
+    with suppress(Exception):
+        comm_address = await get_scheduler_address(
+            scheduler_service_name,
+            namespace,
+            allow_external=False,
+        )
+        async with rpc(comm_address) as scheduler_comm:
+            idle, idle_since = await scheduler_comm.check_idle()
```

Comment:
I don't think this method on the scheduler is returning a tuple right now?

Reply:
Yeah I had done this in my local copy and then went down a different path. I need to update this PR to reflect the changes that landed in distributed. I'm partly waiting for a release to I can bump the minimum version.